nexus/src/internal_api/http_entrypoints.rs

```diff
@@ -591,3 +603,187 @@ async fn ipv4_nat_changeset(
     };
     apictx.internal_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
+
+// APIs for managing blueprints
+//
+// These are not (yet) intended for use by any other programs.  Eventually, we
+// will want this functionality part of the public API.  But we don't want to
+// commit to any of this yet.  These properly belong in an RFD 399-style
+// "Service and Support API".  Absent that, we stick them here.
+
+/// Lists blueprints
+#[endpoint {
+    method = GET,
+    path = "/deployment/blueprints/all",
```

Comment:
So right now it's:

```
# list all blueprints
GET /deployment/blueprints/all
# fetch one blueprint
GET /deployment/blueprints/all/$id
# do other stuff, like get the target
GET /deployment/blueprints/target
POST /deployment/blueprints/regenerate
...
```

Basically, if we put the list endpoint at `/deployment/blueprints`, I'd want to put the individual blueprints at `/deployment/blueprints/$id`, but that overlaps with `/deployment/blueprints/target` (and other blueprint-related stuff we might need).  We avoid that (and dropshot doesn't support it, although only because we assumed we'd want to avoid that :)).

I thought we had done this elsewhere in the public API when we need additional control endpoints but I don't actually see any...

Another way to phrase it could be:

```
# list blueprints
GET /deployment/blueprints
# fetch one blueprint
GET /deployment/blueprints/$id
# do other stuff, like get the target
GET /deployment/target-blueprint
POST /deployment/regenerate-blueprint
```

I could go either way.  I'm not too worried at this point as I don't think there will be programmatic consumers of the internal API aside from omdb.

Reply:
I think you should keep it as is. Thanks for the explanation!